src/lvl_script.c

```diff
@@ -1367,6 +1367,67 @@ static void create_effect_process(struct ScriptContext *context)
     }
 }
 
+static void set_heart_health_check(const struct ScriptLine *scline)
+{
+    ALLOCATE_SCRIPT_VALUE(scline->command, 0);
+    value->arg0 = scline->np[0];
+    if (scline->np[1] > (signed long)game.dungeon_heart_health)
+    {
+        SCRPTWRNLOG("Value %ld is greater than maximum: %ld", scline->np[1], game.dungeon_heart_health);
+        value->arg1 = game.dungeon_heart_health;
+    }
+    else
+    {
+        value->arg1 = scline->np[1];
+    }
+    PROCESS_SCRIPT_VALUE(scline->command);
+}
+
+static void set_heart_health_process(struct ScriptContext *context)
+{
+    struct Thing* heartng = get_player_soul_container(context->value->arg0);
+    if (!thing_is_invalid(heartng))
+    {
+        heartng->health = (short)context->value->arg1;
+    }
+}
+
+static void add_heart_health_check(const struct ScriptLine *scline)
+{
+    ALLOCATE_SCRIPT_VALUE(scline->command, 0);
+    value->arg0 = scline->np[0];
+    value->arg1 = scline->np[1];
+    value->arg2 = scline->np[2];
+    PROCESS_SCRIPT_VALUE(scline->command);
+}
+
+static void add_heart_health_process(struct ScriptContext *context)
+{
+    struct Thing* heartng = get_player_soul_container(context->value->arg0);
+    if (!thing_is_invalid(heartng))
+    {
+        short old_health = heartng->health;
+        long new_health = heartng->health + context->value->arg1;
+        if (new_health > (signed long)game.dungeon_heart_health)
+        {
+            SCRPTWRNLOG("Player %d's calculated heart health (%ld) is greater than maximum: %ld", heartng->owner, new_health, game.dungeon_heart_health);
```

Comment:
I don't think a warning is applicable here. You could for example make a script that loops and give the hearth 1000 health each time an objective is reached, and the player could reach more objectives than his heart needs healing.
Heal spell on creatures also use without error messages.

So I think replacing this with a regular log message which is invisible on default level would be the best approach.

I will update the scripting documentation to make clear that both this (and SET_...) are used to heal/hurt a heart, not change the max_health value.

Reply:
How do I do that?

Reply:
`SCRIPTDBG(7,` instead of `SCRPTWRNLOG(`